src/update/lib/cron.js

```diff
@@ -1,47 +1,70 @@
-/* eslint-disable no-restricted-syntax */
 const { CronJob } = require('cron');
-const fs = require('fs-extra');
-const path = require('path');
+
 const logger = require('./logger');
 
-const reportsDir = path.resolve(__dirname, '..', 'data', 'reports');
-const snapshotDir = path.resolve(__dirname, '..', 'data', 'snapshots');
-const states = path.resolve(__dirname, '..', 'data', 'states');
+const Cron = class Cron {
+  constructor(name, time, task) {
+    this.name = name;
+    this.time = time;
+    this.task = task;
+    this.status = false;
+    this.process = new CronJob(time, async () => {
+      await task();
+    }, null, false, 'Europe/Paris');
+  }
+
+  getConfig() {
+    return {
+      name: this.name,
+      time: this.time,
+      status: this.status,
+    };
+  }
 
-const deleteFilesInDir = async (directory, maxAgeInDays) => {
-  const time = 1 * 24 * 60 * 60 * 1000 * maxAgeInDays;
-  const threshold = Date.now() - time;
+  setTask(task) {
+    this.process.stop();
+    this.task = task;
+    logger.info(`[cron ${this.name}] config - task updated`);
+    this.process = new CronJob(this.time, async () => {
+      await this.task();
+    }, null, false, 'Europe/Paris');
```

Comment:
Unnecessary arrow function, unless you want that the cron job uses `this.task` dynamically instead of using the function given to `setTask()`. But that's not the way it's used in the constructor. Choose either solution but be consistent.

```suggestion
    this.process = new CronJob(this.time, task, null, false, 'Europe/Paris');
```